cylc/flow/task_proxy.py

```diff
@@ -297,6 +301,26 @@ def get_point_as_seconds(self):
                 self.point_as_seconds += utc_offset_in_seconds
         return self.point_as_seconds
 
+    def get_clock_trigger_time(self, offset_str):
```

Comment:
This feels like it might be nice to unit test it: It would make fixing any future issues with this easier.